c2rust-transpile/src/c_ast/conversion.rs

```diff
@@ -1261,7 +1261,11 @@ impl ConversionContext {
                 }
 
                 ASTEntryTag::TagStringLiteral if expected_ty & (EXPR | STMT) != 0 => {
-                    let ty_old = node.type_id.expect("Expected expression to have type");
+                    // https://github.com/immunant/c2rust/issues/1332
+                    // string literals from assert message like this expression
+                    // _Static_assert(1, "message")
+                    //  don't have type. so we give it zero for now
```

Comment:
What does 0 mean for a type ID?

Reply:
I admit this is little hackish😅. I just used least likely used type id which not used by anything (zero). For now because we just skipping transpiling _static_assert, this is fine, but when someday we try to properly to convert `_Static_assert` to `assert!()` or `debug_assert()`, this code will causing panic when accessing the type. 
The right fix is i think we need to give ASTEntryTag::TagStringLiteral some dummy type from C++ side in `ast-exporter`, maybe tonight i will try to familiarize with CLang/LLVM API first

Reply:
I changed the code so we don't need to use this hack. So `StaticAssertDecl` will only visit first parameter, for second parameter, because second parameter of assert is always string literal we just extract the string message and store it directly in CDeclKind::StaticAssert